android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -410,9 +411,7 @@ private void syncComplete() {
 
                         // Event successfully loaded, set data downloaded to true
                         sharedPreferences.edit().putBoolean(ConstantStrings.IS_DOWNLOAD_DONE, true).apply();
-
-                        getDaysBetweenDates(
-                                ISO8601Date.getDateObject(event.getStart()), ISO8601Date.getDateObject(event.getEnd()))
+                        getDaysBetweenDates(ISO8601Date.getTimeZoneDateFromString(event.getStart(), ISO8601Date.getTimeZoneOfEvent()), ISO8601Date.getTimeZoneDateFromString(event.getEnd(), ISO8601Date.getTimeZoneOfEvent()))
```

Comment:
You can create a `DateUtils` method getDateFromString which will be timezone aware and automatically call getTimeZoneofEvent in its implementation. 

Views should just straightforward call the getter and the DateUtils class should handle everything itself without views having to worry about timezone and other things